src/java_bytecode/jar_file.cpp

```diff
@@ -35,11 +38,36 @@ void jar_filet::open(const std::string &filename)
 
   if(mz_ok)
   {
+    // '@' signals file reading with list of class files to load
+    bool regex_match=java_cp_include_files[0]!='@';
+    std::regex regex_matcher;
+    std::smatch string_matcher;
+    std::unordered_set<std::string> set_matcher;
+    jsont json_cp_config;
+    if(regex_match)
+      regex_matcher=std::regex(java_cp_include_files);
+    else
+    {
+      assert(java_cp_include_files.length()>1);
+      if(parse_json(
+           java_cp_include_files.substr(1),
+           get_message_handler(),
+           json_cp_config))
+        throw "cannot read JSON input configuration for JAR loading";
+      assert(json_cp_config.is_object() && "JSON has wrong format");
```

Comment:
This is not an internal error (it's invalid user input), I think - should be throw instead of assert.